### ISSUE TITLE
git.latest rev=tag fails with old git(1)

### ISSUE THREAD

**Issue by orymate** [saltstack/salt#30999]
According to the documentation, from 2015.8 git.latest tries to unset the upstream branch if the specified rev is a tag. This is achieved with the `--unset-upstream` switch of `git branch`, which is unavailable before git 1.9.2. The default git package of Ubuntu 12.04 LTS version is bellow this.

Workaround suggested by @babilen is to add a patched `_states/git.py` with `s/\['--unset-upstream']/None/'` applied.

Reproducing:
```
# salt precise state.low '{"state": "git", "fun": "latest", "name": "https://github.com/saltstack/pepper.git", "rev": "0.1.0", "target": "/tmp/a"}'
precise:
    ----------
    __run_num__:
        0
    changes:
        ----------
        new:
            https://github.com/saltstack/pepper.git => /tmp/a
    comment:
        error: unknown option `unset-upstream'
        usage: git branch [options] [-r | -a] [--merged | --no-merged]
        [...]
            --merged <commit>     print only merged branches
        
        Changes already made: https://github.com/saltstack/pepper.git cloned to /tmp/a. Repository was reset to 0.1.0 (d7c2a24).
    duration:
        3434.94
    name:
        https://github.com/saltstack/pepper.git
    result:
        False
    start_time:
        15:51:08.117198

```

**Comment by jfindlay** [comment]
@orymate, thanks for reporting.

**Comment by terminalmage** [comment]
This is fixed in #31301. Just FYI, I did some digging and found that ``--unset-upstream`` was added in 1.8.0, not 1.9.2.

**Comment by babilen** [comment]
Very nice, ta @terminalmage
